src/OAuthClient/HoldRequestClient.php

```diff
@@ -59,56 +63,88 @@ public static function validateProcessed($processed)
     public static function validateSuccess($success)
     {
         if (!isset($success)) {
-            throw new APIException(
-                'Success flag not set',
-                'Success flag is not set.',
-                0,
+            throw new NonRetryableException(
+                'Not Acceptable: Success flag not set',
+                'Not Acceptable: Success flag is not set.',
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.